src/common/transformations/src/transformations/mlir/op/transpose.cpp

```diff
@@ -27,11 +27,19 @@ struct ConvertTranspose {
         const auto ov_output_shape = node->get_output_partial_shape(0);
         auto out_type = importTensor(context.context, ov_output_shape, ov_output_element_type);
         auto dynamic_dimensions = context.get_dynamic_dimension_values(ov_output_shape);
+        const auto ov_order_element_type = node->get_input_element_type(1);
 
         auto const_order = dynamic_cast<ov::op::v0::Constant*>(node->get_input_node_ptr(1));
         assert(const_order && "non-const order not supported");
-        ov::Coordinate coords = const_order->get_coordinate_val();
-        SmallVector<int64_t> order(coords.begin(), coords.end());
+        SmallVector<int64_t> order;
+        if (ov_order_element_type == ov::element::i64) {
+            ov::Coordinate coords = const_order->get_coordinate_val();
```

Comment:
Indices in `const_order` as the second input of `Transpose` cannot be negative, so you can just use `const_order->get_vector<uint64_t>()` and leave negative indices as UB.

Reply:
Right, I tried that, but got only partial result for some reason. Half of the values were correct, others - zeroes.

Reply:
Make sure that you used `cast_vector`, not `get_vector`.

Reply:
Ah, I see